doc/howtos/image-configuration.rst

```diff
@@ -0,0 +1,100 @@
+default configuration files.
+
+As explained in `yocto-compatible.rst <yocto-compatible.rst>`_, the
+``.bbappend`` files must avoid generic signature changes. So, in order
+to change generic Refkit configuration of a given recipe, you should use
+
+.. code::
+
+    do_install_append_df-refkit-config () {
+        ...
+    }
+
+for appending to the ``do_install()`` phase.
+
+Using virtual packages
+======================
+
+Some recipes in Refkit and other Yocto layers have the configuration
+split into separate configuration packages. For example, see
+`nftables-settings-default recipe
+<../../meta-refkit-core/recipes-security/nftables-settings-default/nftables-settings-default_0.1.bb>`_.
+In these cases, if you want to replace the configuration fully, you can
+create your own configuration package (say ``nftables-settings-custom``)
+and just use it in place of the default configuration package.
+
+Other recipes, where having a configuration package is mandatory, take
+this further and use ``VIRTUAL_RUNTIME`` convention to enforce that a
+configuration package will be installed to the target. For example,
+consider `groupcheck recipe
+<../../meta-refkit-core/recipes-security/groupcheck/groupcheck_git.bb>`_,
+which has the following lines:
+
+.. code::
+
+    VIRTUAL-RUNTIME_groupcheck-settings ?= "groupcheck-settings-default"
+    RDEPENDS_${PN} += "${VIRTUAL-RUNTIME_groupcheck-settings}"
+
+If a configuration package (say ``groupcheck-settings-custom``) is made,
+it needs to be set to be the required configuration package. This is
+done by changing ``local.conf`` or other distro configuration file to
+override the particular ``VIRTUAL_RUNTIME`` variable:
+
+.. code::
+
+    VIRTUAL-RUNTIME_groupcheck-settings = "groupcheck-settings-custom"
+
+Image configuration during root filesystem creation
+===================================================
+
+An alternative to configuring individual packages is the configuration
+of the entire image during rootfs creation. The image recipes can be
+appended as any other recipe. This is the correct approach if you have
+several files in different packges which you need to change or if you
+just want to keep the configuration changes in one place.
+
+One practical way for implementing a certain configuration would be to
+add it as a ``.bbclass`` file, which would then be inherited by all
+images which require that configuration. For example, you can create
+``custom-config.bbclass`` file and add a Python task function to do the
+necessary configuration changes there:
+
+.. code::
+
+    python change_image_configuration () {
+        etcdir = oe.path.join(d.getVar("IMAGE_ROOTFS"), d.getVar("sysconfdir"))
+        ...
+    }
+
+    IMAGE_PREPROCESS_COMMAND_prepend = "change_image_configuration;"
+
+You can either inherit the file directly into your own image file or
+alternatively add to your distro configuration file
+
+.. code::
+
+    INHERIT_refkit-image-<profilename> += "custom-config.bbclass"
```

Comment:
Are you sure that this works?

I suspect you meant:

    INHERIT_append_pn-refkit-image-<profilename> = " custom-config"

Note that this (currently) has the same problem with the virtual refkit-image-<profilename>-flatpak as with the virtual swupd images: because the PN is different in the virtual images, the override is only applied to the base image.

I had (finally, I should probably say) and idea how to address that: when creating the virtual image, we can simply extend the OVERRIDES to include not just PN, but also the base PN. However, meta-flatpak uses image-variant.bbclass, which currently cannot know whether that OVERRIDES change is desired, so the actual implementation will be a bit more difficult.